src/bindings/c/tests/ov_core_test.cpp

```diff
@@ -99,24 +99,49 @@ TEST_P(ov_core, ov_core_compile_model) {
     ASSERT_NE(nullptr, model);
 
     ov_compiled_model_t* compiled_model = nullptr;
-    ov_properties_t* property = nullptr;
-    OV_ASSERT_OK(ov_core_compile_model(core, model, device_name.c_str(), property, &compiled_model));
+    OV_ASSERT_OK(ov_core_compile_model(core, model, device_name.c_str(), false, &compiled_model));
     ASSERT_NE(nullptr, compiled_model);
 
     ov_compiled_model_free(compiled_model);
     ov_model_free(model);
     ov_core_free(core);
 }
 
+TEST_P(ov_core, ov_core_compile_model_with_property) {
+    auto device_name = GetParam();
+    ov_core_t* core = nullptr;
+    OV_ASSERT_OK(ov_core_create(&core));
+    ASSERT_NE(nullptr, core);
+
+    ov_model_t* model = nullptr;
+    OV_ASSERT_OK(ov_core_read_model(core, xml, nullptr, &model));
+    ASSERT_NE(nullptr, model);
+
+    ov_compiled_model_t* compiled_model = nullptr;
+    const char* key = ov_property_key_num_streams;
+    const char* type = ov_property_value_type_uint32;
+    uint32_t num = 11;
+    OV_ASSERT_OK(ov_core_compile_model(core, model, device_name.c_str(), true, &compiled_model, key, type, num));
```

Comment:
I thought we will have API the same as in C++: variadic function for: `key1, value1, key2, value2` 

@ilya-lavrenov What do you think?

Reply:
yes, but in C maybe we still need the parameter to indicate the number of variadic arguments, I cannot find a safe solution to figure out it.

Reply:
Here, I don't think that we need to provide a type for variables.
In legacy API we use only two string key and value, here you add a new variable: type.

Reply:
So we can remove type and make all values as string type, then pass them to ov::Any to convert?

Reply:
Done - remove type variable.